### PR TITLE
Update packaging and publishing instructions

### PR DESCRIPTION
**Pull request by stevemchugh**

### CODE REVIEW

File: nuget/nuget.md [hunk 1]
```diff
@@ -152,120 +152,152 @@ Set-ExecutionPolicy Unrestricted -Scope Process
...
 
-The general convention for naming Pre Release packages is to suffix the package with -pre
+Once you have created your Team Project under https://tfs.burrows.biz/tfs/NuGet/, there are a number of steps required to prepare the Team Project for packaging and publishing the NuGet package(s).
 
-e.g. burrows.testing.1.0.0-pre
+1. Setup the usual Pull Request Build, Continuous Integration Build, branch policy on develop branch and Slack integration hooks.
+2. On your Team Project's admin page, select the Services tab, and add a new Generic Service Endpoint
+3. On the Add New Generic Endpoint dialog set the:
+**Connection Name** to "Burrows NuGet"
+**Server URL** to "https;//nuget.burrows.biz/"
```
Comment (petefitz) [inline nuget/nuget.md]: change semi-colon to colon

File: nuget/nuget.md [hunk 2]
```diff
@@ -152,120 +152,152 @@ Set-ExecutionPolicy Unrestricted -Scope Process
...
+**User name** leave blank
+**Password/Token Key** to the **[API Key](http://tfs.burrows.biz/tfs/NuGet/_git/nuget.burrows.biz#path=%2Freadme.md&version=GBmaster&_a=contents)** for "https;//nuget.burrows.biz/"
```
Comment (petefitz) [inline nuget/nuget.md]: change semi-colon to colon

File: nuget/nuget.md [hunk 3]
```diff
@@ -152,120 +152,152 @@ Set-ExecutionPolicy Unrestricted -Scope Process
...
+### Team Project Setup
 
-The general convention for naming Pre Release packages is to suffix the package with -pre
+Once you have created your Team Project under https://tfs.burrows.biz/tfs/NuGet/, there are a number of steps required to prepare the Team Project for packaging and publishing the NuGet package(s).
 
-e.g. burrows.testing.1.0.0-pre
+1. Setup the usual Pull Request Build, Continuous Integration Build, branch policy on develop branch and Slack integration hooks.
+2. On your Team Project's admin page, select the Services tab, and add a new Generic Service Endpoint
+3. On the Add New Generic Endpoint dialog set the:
+**Connection Name** to "Burrows NuGet"
```
Comment (petefitz) [inline nuget/nuget.md]: Can this be a table or a screenshot - viewing the md file it looks a bit squished

File: nuget/nuget.md [hunk 4]
```diff
@@ -152,120 +152,152 @@ Set-ExecutionPolicy Unrestricted -Scope Process
...
 ```
 
-If the version number ends in -text (e.g. 1.0.0-alpha) then the package will be a -pre package.
+Take note of the replacement values e.g. $id$. These values are merged from the C# project properties. They only need to be maintained on the C# project. Release notes and tags are optional but, tags are especially useful for searching for packages.
 
-For more info on nuget.exe: https://docs.nuget.org/consume/command-line-reference
-Note, Nuget.exe needs to live in your %Path% but for some reason cannot live in the /system32/ folder.  Best bet is to add it to C:/Windows.
-- - -
-## Publishing Packages
+The icon URL is for a correctly sized Burrows Lozenge 64x64 pixels on a transparent png. this is the correct file type and size for nuget packages and should be used by default.
```
Comment (petefitz) [inline nuget/nuget.md]: Capitalise "this is the correct ..."

File: nuget/nuget.md [hunk 5]
```diff
@@ -152,120 +152,152 @@ Set-ExecutionPolicy Unrestricted -Scope Process
...
+
+Setup the Visual Studio Build Step to build the solution as normal
+
+![](media/build-step-1.jpg)
+
+Setup the Visual Studio test Step to run the tests as normal
+
+![](media/build-step-2.jpg)
+
+Add a NuGet Packager step and set the Path/pattern to nuspec files as the path to the .csproj file(s) you want to package. Ensure you untick the Use Buildnumber to version package check box. Under Advanced, ensure you use the $(BuildConfiguration) variable to specify the Configuration to package
```
Comment (petefitz) [inline nuget/nuget.md]: put apostrophes round the 'checkbox text'

File: nuget/nuget.md [hunk 6]
```diff
@@ -152,120 +152,152 @@ Set-ExecutionPolicy Unrestricted -Scope Process
...
+
+![](media/build-step-3.jpg)
+
+Add a NuGet Publisher step and set the Path/Pattern to nupkg to a wildcard pattern that restricts the step to picking up your package(s) only. This is needed because it can also pickup dependent packages that are pulled down as part of the build, and we ae not interested in publishing those to our internal nuget. Set the NuGet Server Endpoint to the one that was created earlier in this tutorial.
```
Comment (petefitz) [inline nuget/nuget.md]: we are not interested

File: nuget/nuget.md [hunk 7]
```diff
@@ -152,120 +152,152 @@ Set-ExecutionPolicy Unrestricted -Scope Process
...
+
+![](media/build-step-4.jpg)
+
+Final steps...
 
-* Add references from your .NuGet project to the projects that need to be included in the nuget package.
-* Add a file called NuGetPack.config to your Package Project:
+Under the Repository tab, default the Repository to the develop branch
```
Comment (petefitz) [inline nuget/nuget.md]: new lines not showing in github md view

File: nuget/nuget.md (hunk 1 above)
Comment (stevemchugh) [inline nuget/nuget.md]: Done

File: nuget/nuget.md (hunk 2 above)
Comment (stevemchugh) [inline nuget/nuget.md]: Done

File: nuget/nuget.md (hunk 3 above)
Comment (stevemchugh) [inline nuget/nuget.md]: Done

File: nuget/nuget.md (hunk 4 above)
Comment (stevemchugh) [inline nuget/nuget.md]: Done

File: nuget/nuget.md (hunk 5 above)
Comment (stevemchugh) [inline nuget/nuget.md]: Done

File: nuget/nuget.md (hunk 6 above)
Comment (stevemchugh) [inline nuget/nuget.md]: Done

File: nuget/nuget.md (hunk 7 above)
Comment (stevemchugh) [inline nuget/nuget.md]: Done

**Comment by petefitz** [comment]
LGTM

File: nuget/nuget.md [hunk 8]
```diff
@@ -152,120 +153,156 @@ Set-ExecutionPolicy Unrestricted -Scope Process
...
+######Project Properties
 
-or drop your nuget package directly on to the share  on the server
+Set the Assembly name to something that represents the component, this will also become the nuget package name and is merged to the nuspec file as the id. Typically this would also be the default namespace of the component.
 
-> [\\\bur5tfs01\nuget-packages-repository](\\bur5tfs01\nuget-packages-repository)
+![](media/example-project-properties.jpg)
 
-You will need to be a member of the EMEA-WUN-BUR-5-Developers group to be able to do this
+Set the Assembly Information, this will be merged with the nuspec file when packing the nuget package. The replacement values set in the nuspec file will mainly be pulled from here. A good level of metadata to set would be:
```
Comment (Phoooey) [inline nuget/nuget.md]: I think this should be Click on the Assembly Information button .....

File: nuget/nuget.md [hunk 9]
```diff
@@ -152,120 +153,156 @@ Set-ExecutionPolicy Unrestricted -Scope Process
...
+    <language>en-US</language>
+    <projectUrl>https://tfs.burrows.biz/tfs/NuGet/your-team-project-name/</projectUrl>
+    <iconUrl>https://nuget.burrows.biz/burrows_64.png</iconUrl>
+    <tags></tags>
+  </metadata>
+</package>
 ```
 
-If the version number ends in -text (e.g. 1.0.0-alpha) then the package will be a -pre package.
+Take note of the replacement values e.g. $id$. These values are merged from the C# project properties. They only need to be maintained on the C# project. Release notes and tags are optional but, tags are especially useful for searching for packages.
```
Comment (Phoooey) [inline nuget/nuget.md]: I think it should be
Take note of the values e.g. $id$,  these values are merged from the C# project properties. Release notes and tags are optional but, tags are especially useful for searching for packages.

File: nuget/nuget.md (hunk 8 above)
Comment (stevemchugh) [inline nuget/nuget.md]: Ok

File: nuget/nuget.md (hunk 9 above)
Comment (stevemchugh) [inline nuget/nuget.md]: Re-done, incorporated your feedback

**Pull request merged**